### PR TITLE
Unify query and url-encoded-form implementation

### PR DESCRIPTION
**Pull request by crazytonyli**
### Description

There are two changes in this PR.

#### URL encoding in query and form

We used to have different implementation around encoding name-value pair into URL query and web form in POST request body: `URLComponent`'s query encoding implementation is used in building URL query, and our own customized implementation (which aligns with Alamofire's implementation for backwards compatibility) is used to build web form.

Now, the same implementation, which is our customized one, is used to build both query and web form. Backwards compatibility is the only reason for this change.

#### HTTPRequestBuilder now does not modify the original URL

Say we have `HTTPRequestBuilder` instance created using an URL "https://wp.org/api?locale=en". When adding a query `.query(name: "locale", value: "zh")`, the new query replaces the one in the original url. This behaviour is changed in this PR. The added query will be appended onto the original URL, which means the original url content will always be in the final result. To be honest, I don't like this new behavior, but that's what Alamofire does and we'll need to do it for backwards compatibility.

### Testing Details

See added unit tests.

---

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

### CODE REVIEW

File: WordPressKitTests/Utilities/HTTPRequestBuilderTests.swift [hunk 1]
```diff
@@ -5,6 +5,43 @@ import XCTest
 
 class HTTPRequestBuilderTests: XCTestCase {
 
+    static let nestedParameters: [String: Any] =
+        [
+            "number": 1,
+            "nsnumbe-true": NSNumber(value: true),
```
Comment (jkmassel) [inline WordPressKitTests/Utilities/HTTPRequestBuilderTests.swift]: nit: typo?

File: WordPressKit/HTTPRequestBuilder.swift [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import Foundation
 
+/// A builder type that adding HTTP request data onto an URL.
```
Comment (jkmassel) [inline WordPressKit/HTTPRequestBuilder.swift]: nit: typo?

**Review by jkmassel — APPROVED** [review]
Little things, the rest we discussed on Zoom

File: WordPressKit/HTTPRequestBuilder.swift (hunk 2 above)
Comment (crazytonyli) [reply WordPressKit/HTTPRequestBuilder.swift]: 🙈 Fixed in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/694/commits/59bd717519e4ce71a4091193bd2015ea59454607.

**Pull request merged**